### PR TITLE
add lock function

### PR DESCRIPTION
**Pull request by HanshaLeng**
Provide a general summary of your changes in the Title of the PR

**Description**

Describe your changes in detail. Please use present tense without a subject to describe your PR, for example: "Fix**es** issue 123" or "Adds translations of levels 1 to 12 for Polish"

**Fix for**

Describe what issue or discussion your pr concerns

**How to test**

Describe how to run your code to see it is action. See this https://github.com/Felienne/hedy/pull/880#issue-1016304308 for an example.

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [ ] Links to an existing issue or discussion (if not, create an issue first)
- [ ] Describes changes clear in the format above (present tense, no subject)
- [ ] Has a "how to test" section
- [ ] Only one thing is done in this pull request (specifically please try to refrain from mixing textual changes to the yamls from code changes)

### CODE REVIEW

**Comment by Felienne** [comment]
Hi @HanshaLeng!

Thanks for your PR, I am seeing some issues, can you look into those? See: https://github.com/Felienne/hedy/runs/4168319214?check_suite_focus=true

You can also try to run these tools on your own machine to try to investigate the issue, they are in buildtools/github

Good luck!

File: grammars/level8-1-Additions.lark [hunk 1]
```diff
@@ -0,0 +1,4 @@
+command: print | ifs elses? | ask | for_loop | assign_list | list_access_var | assign
+
+//new for level 8-1, for loop and the end-blocks
+for_loop: _FOR _SPACE (NAME | var) _SPACE _IS _SPACE (NUMBER | var) _SPACE _TO _SPACE (NUMBER | var) _EOL (_SPACE command) (_EOL _SPACE command)* _EOL "end-block"
```
Comment (bjorn3) [inline grammars/level8-1-Additions.lark:4]: I believe this level was removed somewhat recently.

File: build-tools/validate-e2e [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (bjorn3) [inline build-tools/validate-e2e:1]: A pretty much identical file exists at build-tools/github/validate-e2e. The only difference is that this file has less debug output.

File: grammars/level8-1-Additions.lark (hunk 1 above)
Comment (Felienne) [reply grammars/level8-1-Additions.lark:4]: YEs this PR branches of a very old version of main, a new one is in the works

**Pull request closed without merging**